app/views/checkout/thankyou.scala.html

```diff
@@ -41,9 +47,29 @@ <h3 class="section-header">Download your apps</h3>
                 <li>Open the app and tap the 'Settings' icon</li>
                 <li>Enter your subscription code (shown above and also in your confirmation email) and your postcode</li>
             </ol>
+
+        @formOpt.map { form =>
             <h3 class="section-header">Register for an account</h3>
             <p>Some features of your subscription require you to be signed in before you can make use of them.</p>
-            <a href="#" class="button button--primary button--large">Register now</a>
+            <p>Enter a password to finish creating your account:</p>
+            <form action="@processFinishAccount" method="POST" class="form">
+                <fieldset class="fieldset">
+                    <div class="fieldset__fields">
+                        <div class="form-field js-checkout-finish-account-password">
+                            <label class="label" for="last">Password</label>
+                            <input name="@userIdParam" type="hidden" value="@form.data.get(userIdParam)">
+                            <input name="@identityTokenParam" type="hidden" value="@form.data.get(identityTokenParam)">
+                            <input name="password" type="password" class="input-text js-input">
+                            @checkout.errorMessage("Password is too short")
```

Comment:
We should list what the requirements for password are somewhere (here or an FAQ page?). cc @rtyley 
@afiore Do we get error messages from Identity as perhaps we can pass this through here?